server/routes/farcast/swap.ts

```diff
@@ -0,0 +1,69 @@
+import router from './index.js'
+import { authMessage, getPageSetting } from './middlewares.js'
+import config from '../../config.js'
+import { lookupFid, renderImageResponse } from '../../src/farcaster.js'
+import type { Request } from 'express'
+
+const generateImageResponse = (text: string, req: Request, retry = false): string => {
+  const host = req.get('host')
+  const image = `${req.protocol}://${host}/${config.farcast.apiBase}/text/image?t=${encodeURIComponent(text)}`
+  const restartTarget = `${req.protocol}://${host}/${config.farcast.apiBase}/callback/redirect`
+
+  if (retry) {
+    return renderImageResponse(image, 'Retry', 'post', restartTarget)
+  }
+
+  return renderImageResponse(image)
+}
+
+router.post('/swap/buy/callback', authMessage, getPageSetting, async (req, res) => {
+  if (!req.validatedMessage) {
+    const html = generateImageResponse('Failed to validate the message. Please try again later.', req, true)
+    return res.send(html).end()
+  }
+
+  const balanceInSat = 0
+  const price = 0
+
+  // TODO: BUY
+
+  const fid = req.validatedMessage.data?.fid
+
+  if (!fid) {
+    console.error('[/farcast/buy/callback] No fid found in validatedMessage')
+    const html = generateImageResponse('Failed to get fid. Please try again later.', req, true)
+    return res.send(html).end()
+  }
+  const { owner } = await lookupFid(fid)
```

Comment:
could fail, check and use failure route

Reply:
Sorry, I can't see fail handler in the other extension when getting owner from `lookupFid()`
Want to put it in try / catch and handle error in catch?

Reply:
okay

Reply:
Done. Put it in try catch clause